projects/js-toolkit/packages/liferay-cli/src/new/facet-portlet/index.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * SPDX-FileCopyrightText: © 2017 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+import {
+	FilePath,
+	TRANSFORM_OPERATIONS,
+	TemplateRenderer,
+	format,
+	transformJsonFile,
+	transformTextFile,
+} from '@liferay/js-toolkit-core';
+import fs from 'fs';
+
+import getPortletName from '../util/getPortletName';
+import prompt from '../util/prompt';
+
+import type {Facet, Options} from '../index';
+
+const {
+	PkgJson: {addPortletProperties},
+	Text: {appendLines},
+} = TRANSFORM_OPERATIONS;
+const {info, print} = format;
+
+const facet: Facet = {
+	async prompt(useDefaults: boolean, options: Options): Promise<Options> {
+		return await prompt(useDefaults, options, [
+			{
+				default: 'category.sample',
+				message: 'Under which category should your widget be listed?',
+				name: 'category',
+				type: 'input',
+			},
+		]);
+	},
+
+	async render(options: Options): Promise<void> {
+		const renderer = new TemplateRenderer(
+			new FilePath(__dirname).join('templates'),
+			options.outputPath
+		);
+
+		await renderer.render('assets/css/styles.css', options);
+
+		// Add portlet properties
+
+		const pkgJsonFile = options.outputPath.join('package.json');
+
+		/* eslint-disable-next-line @liferay/liferay/no-dynamic-require, @typescript-eslint/no-var-requires */
+		const pkgJson = require(pkgJsonFile.asNative);
+		const portletDisplayName = pkgJson['name'];
+		const portletName = getPortletName(portletDisplayName);
+
+		print(info`Generating porlet...`);
+		print(info`  Setting porlet name: {${portletDisplayName}}`);
+		print(info`  Setting portlet category: {${options.category}}`);
+
+		await transformJsonFile(
+			pkgJsonFile,
+			pkgJsonFile,
+			addPortletProperties({
+				'com.liferay.portlet.display-category': options.category as string,
+				'com.liferay.portlet.header-portlet-css': '/css/styles.css',
```

Comment:
Is there any particular reason we use `styles` vs `main`? I noticed it is sort of split in dxp

Reply:
This also makes me think, should we also add an option for using sass or we could also do that in a future ticket.

Reply:
:thinking: In fact, we had that option in the original generator IINM :thinking: . Don't know how it disappeared...

Let's create a [new issue](https://github.com/liferay/liferay-frontend-projects/issues/589) so that we don't make this PR huge.

Reply:
I used `styles.css` because it was `styles.css` in the previous generator.  But I don't think there's any specific reason for that...

Looking at searches in GH for [styles.css](https://github.com/search?q=styles.css) and [main.css](https://github.com/search?q=main.css) it looks like both names are relevant, so I guess we can go with whichever one you like...